fdbclient/CommitTransaction.h

```diff
@@ -107,7 +107,7 @@ struct MutationRef {
 
 	template <class Ar>
 	void serialize( Ar& ar ) {
-		if (!ar.isDeserializing && type == ClearRange && equalsKeyAfter(param1, param2)) {
+		if (ar.isSerializing && type == ClearRange && equalsKeyAfter(param1, param2)) {
```

Comment:
Why do we want to serialize an uninitiated MutationRef? ( according to "In this case, we are calling serialize with an uninitialized MutationRef," in the PR.)
Would you mind providing more context?

Reply:
flatbuffers determines the schema of an object of type T by calling `T::serialize` on a default constructed instance, local to flatbuffers. That could be changed in the future but that's the current situation. That's the uninitialized MutationRef in question here.

Reply:
And to elaborate, we wouldn't want to serialize an uninitialized `MutationRef`, which is what the `if` guard prevents. It just turns out that `!ar.isDeserializing` isn't the same as `ar.isSerializing` because of the 3rd use-case Andrew describes above. This change more accurately describes that we only want to run the if block if we are in fact serializing the `MutationRef`.